pubsub/pulsar/pulsar.go

```diff
@@ -373,28 +440,36 @@ func (p *Pulsar) Subscribe(ctx context.Context, req pubsub.SubscribeRequest, han
 	go func() {
 		defer p.wg.Done()
 		defer cancel()
-		p.listenMessage(listenCtx, req.Topic, consumer, handler)
+		p.listenMessage(listenCtx, req, consumer, handler)
 	}()
 
 	return nil
 }
 
-func (p *Pulsar) listenMessage(ctx context.Context, originTopic string, consumer pulsar.Consumer, handler pubsub.Handler) {
+func (p *Pulsar) listenMessage(ctx context.Context, req pubsub.SubscribeRequest, consumer pulsar.Consumer, handler pubsub.Handler) {
 	defer consumer.Close()
 
+	originTopic := req.Topic
 	var err error
 	for {
 		select {
 		case msg := <-consumer.Chan():
-			// Go routine to handle multiple messages at once.
-			p.wg.Add(1)
-			go func(msg pulsar.ConsumerMessage) {
-				defer p.wg.Done()
+			if getProcessMode(req) {
```

Comment:
I think returning a boolean from this method is not very clear. Perhaps consider something like `if processMode(req) == async`